prow/tide/tide.go

```diff
@@ -157,6 +163,24 @@ func NewController(ghcSync, ghcStatus *github.Client, kc *kube.Client, ca *confi
 	}
 }
 
+func (b *branchProtection) skipContext(c string) bool {
+	if !b.config.UseRequiredChecksOnly {
+		return false
+	}
+	return !b.requiredChecks.Has(c)
+}
+
+func (b *branchProtection) missingRequiredContext(c sets.String) []Context {
+	if !b.config.UseRequiredChecksOnly {
+		return nil
+	}
+	var ctxs []Context
+	for _, c := range b.requiredChecks.Difference(c).List() {
+		ctxs = append(ctxs, Context{Context: githubql.String(c)})
```

Comment:
Shouldn't construct semi-initialized `Context` object here, someone's going to nil pointer downstream of this

Reply:
done